### DISCUSSION TITLE
Should Dew Heater "Enable Day" - when disabled override threshold settings to keep heater off during day?

### DISCUSSION THREAD

**Discussion by bendenajones** [aaronwmorris/indi-allsky#1555] [General]
I expected that the "Enable Day" switch would override the threshold settings; as in when the Dew Heater is enabled then the threshold settings would apply. Don't want my Dew Heater on during daylight hours but it is often until outside temp exceeds threshold values. It would seem as if it's either or; i.e. thresholds OR daytime switch.

![image](https://github.com/user-attachments/assets/57b1bb72-7c8d-4844-b5eb-e70917b0d8ea)

**Comment by aaronwmorris** [comment]
I suppose I do not understand the question.  The "Enable Day" option just allows the dew heater to operate during the day.  If you enable daytime operation, it still has to obey the threshold options.

**Comment by bendenajones** [comment]
How about if it's disabled? Should the threshold settings have any effect then? Or perhaps I am confused about when "day time" starts. This morning at ~8am local time, the heater was on; the dew pt and specified sensor temp were within the medium threshold. So I thought that 8am is not night time and why is the heater on?

**Reply by aaronwmorris** [reply]
I understand what you are saying now.  I finally hooked up my oscilloscope to see the pin operations and there is obviously some logic errors when thresholds are enabled.

**Reply by bendenajones** [reply]
let me know when a fix is in and I'll pull it down and test. thanks.

**Reply by aaronwmorris** [reply]
I merged #1583 earlier today and forgot to update here.

**Comment by bendenajones** [comment]
Working as expected now with fix. ty
